client/lib/browser-storage/index.ts

```diff
@@ -32,7 +34,18 @@ const getDB = once( () => {
 					}
 					reject( request.error );
 				};
-				request.onsuccess = () => resolve( request.result );
+				request.onsuccess = () => {
+					const db = request.result;
+					// Add a general error handler for any future requests made against this db handle.
+					// See https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API/Using_IndexedDB#Handling_Errors for
+					// more information on how error events bubble with IndexedDB
+					db.onerror = function( errorEvent: any ) {
+						if ( errorEvent.target?.error?.name ) {
+							mc.bumpStat( 'calypso-browser-storage', kebabCase( errorEvent.target.error.name ) );
```

Comment:
If we're interested in the error traces also logging to kibana may make sense. We could sample the logging if there are many events.